composer/callbacks/mlperf.py

```diff
@@ -0,0 +1,285 @@
+import json
+import logging
+import os
+import platform
+import sys
+import warnings
+from typing import Dict, Optional
+
+import cpuinfo
+import psutil
+import torch
+from mlperf_logging import mllog
+
+import composer
+from composer import Callback, State
+from composer.loggers import Logger
+from composer.utils import dist
+
+try:
+    from mlperf_logging import mllog
+    from mlperf_logging.mllog import constants
+    mlperf_available = True
+except ImportError:
+    mlperf_available = False
+
+BENCHMARKS = ("resnet")
+DIVISIONS = ("open")
+STATUS = ("onprem", "cloud", "preview")
+
+
+def rank_zero() -> bool:
+    return dist.get_global_rank() == 0
+
+
+class MLPerfCallback(Callback):
+    """Creates a compliant results file for MLPerf Training benchmark.
+
+    A submission folder structure will be created with the ``root_folder``
+    as the base and the following directories:
```

Comment:
Preformat the folder structure.

```suggestion
    as the base and the following directories::
```